components/MenuList/ListingItems/ListingItems.tsx

```diff
@@ -1,29 +1,68 @@
 import style from './Listing.module.scss'
 import MenuItemModel from '../../../models/MenuItemModel'
 import Link from 'next/link'
-import { formatPrice } from '../../../utils/dataHelper'
+import { compareWeekDaysToModel, formatPrice } from '../../../utils/dataHelper'
+import { useState, useEffect } from 'react'
+import { getServerDate } from '../../../utils/apiHelper'
 
 interface Props {
    item: MenuItemModel
    tester: boolean
 }
 
+interface LinkRenderProps {
+   children: any
+   link: boolean
+}
+
 export default function Listing({ item, tester }: Props) {
+   const [activeItem, setActiveItem] = useState(compareWeekDaysToModel(item.weekDays))
+
+   const LinkRender = ({ children, link }: LinkRenderProps) => {
```

Comment:
@brunorplima Do you think it's okay for a Component to be maintained inside another component like this or should we separate it?

Reply:
It's good practice to always keep components separated, but when it comes to a component with a very small piece of UI/logic like in this situation I sometimes decide to keep in the same component...
Is this component ever going to be suitable to be used elsewhere in the application? If not it's not a big deal to keep it within another component. A second option for this case is to create another file for the small component inside the ListingItems folder since you're only going to use it here. This second option is a good one if you want your code to be more professional.